riscv/mmu.h

```diff
@@ -481,19 +470,20 @@ class mmu_t
     return (uint16_t*)(translate_insn_addr(addr).host_offset + addr);
   }
 
-  inline trigger_matched_t *trigger_exception(trigger_operation_t operation,
+  inline triggers::matched_t *trigger_exception(triggers::operation_t operation,
       reg_t address, reg_t data)
   {
     if (!proc) {
       return NULL;
     }
-    int match = proc->trigger_match(operation, address, data);
-    if (match == -1)
+    triggers::action_t action;
+    auto match = proc->TM.memory_access_match(&action, operation, address, data);
+    if (match == triggers::MATCH_NONE)
       return NULL;
-    if (proc->state.mcontrol[match].timing == 0) {
-      throw trigger_matched_t(match, operation, address, data);
+    if (triggers::MATCH_FIRE_BEFORE) {
```

Comment:
Shouldn't this compare vs `match` i.e. `if (match == triggers::MATCH_FIRE_BEFORE)`? Otherwise it seems like this will always be taken, since `triggers::MATCH_FIRE_BEFORE` is nonzero.

Reply:
Yes. Good catch.